synapse/storage/database.py

```diff
@@ -412,6 +413,16 @@ def __init__(
                 self._check_safe_to_upsert,
             )
 
+        # We define this sequence here so that it can be referenced from both
+        # the DataStore and PersistEventStore.
```

Comment:
This feels awfully dirty to put this here. Should we have a `PersistEventWorkerStore` that contains just this?

Reply:
It is dirty. I don't think having a shared base class works as we the `PersistEventStore` isn't part of the mixin, and so you'd lead to two instances of the sequence (which is fine for postgres, but not fine for SQLite)